packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -399,23 +406,28 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 					)
 				);
 
-				const middleware = createMiddleware(
-					({ request }) => {
-						return miniflare.dispatchFetch(toMiniflareRequest(request), {
-							redirect: "manual",
-						}) as any;
-					},
-					{ alwaysCallNext: false }
-				);
-
 				handleWebSocket(
 					vitePreviewServer.httpServer,
 					() => miniflare.dispatchFetch
 				);
 
 				// In preview mode we put our middleware at the front of the chain so that all assets are handled in Miniflare
-				vitePreviewServer.middlewares.use((req, res, next) => {
-					middleware(req, res, next);
+				vitePreviewServer.middlewares.use(async (req, res, next) => {
+					try {
+						const request = createRequest(req, res);
+						const response = await miniflare.dispatchFetch(
+							toMiniflareRequest(request),
+							{ redirect: "manual" }
+						);
+
+						if (req.httpVersionMajor === 2) {
+							response.headers.delete("transfer-encoding");
+						}
```

Comment:
I think that it would be pretty valuable to add a comment here explaining why this is needed

Reply:
And above in the dev server

Reply:
[f3d678f](https://github.com/cloudflare/workers-sdk/pull/9152/commits/f3d678f3114c97417ad45a63ea522fd84c7b0c55)